app/configurator/configurator-state.tsx

```diff
@@ -636,15 +639,32 @@ export const canTransitionToPreviousStep = (_: ConfiguratorState): boolean => {
   return true;
 };
 
+// FIXME: should by handled better, as color is a subfield and not actual field.
+// Side effects in ui encodings?
```

Comment:
Why is it important if color is a subfield or an actual field ? From I what I understand, the goal here is to extract the needed filters based on the current IRIs used in the config, and we need special treatment for IRIs used in color mapping because they are not generic. Do I understand correctly ? In this case why does it matter if it's a field or a subfield ?

 Can you add a docstring to getFiltersByMappingStatus so that when we return to it later, it's easier to understand ?

Reply:
I see, thanks for the explanation, I'll try to keep an eye for this and see if I think of a better way :)